types/index.d.ts

```diff
@@ -0,0 +1,491 @@
+  on(event: 'afterError', callback: OktaSIW.EventCallbackWithError): void;
+  on(event: 'afterRender', callback: OktaSIW.EventCallback): void;
+  on(event: 'pageRendered', callback: OktaSIW.EventCallbackWithData): void;
+
+  off(event?: 'ready', callback?: OktaSIW.EventCallback): void;
+  off(event?: 'afterError', callback?: OktaSIW.EventCallbackWithError): void;
+  off(event?: 'afterRender', callback?: OktaSIW.EventCallback): void;
+  off(event?: 'pageRendered', callback?: OktaSIW.EventCallbackWithData): void;
+
+  show(): void;
+  hide(): void;
+  remove(): void;
+
+  showSignInToGetTokens(options: OktaSIW.ShowSignInToGetTokensOptions): Promise<Tokens>;
+  showSignInAndRedirect(options: OktaSIW.ShowSignInAndRedirectOptions): Promise<void>;
+  renderEl(
+    options: OktaSIW.RenderElOptions,
+    success?: (res: OktaSIW.RenderResult) => void,
+    error?: (err: OktaSIW.RenderError) => void
+  ): Promise<OktaSIW.RenderResult>;
+}
+
+export namespace OktaSIW {
+  // Config
+  interface WidgetConfig {
+    // Basic config options
+    baseUrl: string;
+    logo?: string;
+    logoText?: string;
+    helpSupportNumber?: string;
+    brandName?: string;
+    // Username and password
+    username?: string;
+    transformUsername?: (username: string, operation: UserOperation) => string;
+    processCreds?: (creds: Creds, callback?: SimpleCallback) => void;
+    // Language and text
+    language?: LanguageCode | LanguageCallback;
+    i18n?: {
+      [languageCode in LanguageCode]?: {
+        [propertyKey: string]: string,
+      };
+    };
+    assets?: {
+      baseUrl?: string;
+      rewrite?: (assetPath: string) => string;
+    };
+    // Colors
+    colors?: {
+      [colorKey in ColorKey]?: string;
+    };
+    // Links
+    helpLinks?: {
+      help?: string,
+      forgotPassword?: string,
+      factorPage?: LinkBasic,
+      unlock?: string,
+      custom?: Array<Link>,
+    };
+    signOutLink?: string;
+    // Buttons
+    customButtons?: Array<CustomButton>;
+    // Registration
+    registration?: {
+      click?: SimpleCallback;
+    } & Registration.Callbacks;
+    // Feature flags
+    features?: {
+      [featureKey in Feature]?: boolean;
+    };
+    // OIDC
+    clientId?: string;
+    redirectUri?: string;
+    authParams?: AuthParams;
+    authClient?: OktaAuth;
+    idps?: Array<Idp | IdpLink>;
+    idpDisplay?: IdpDisplay;
+    oAuthTimeout?: number;
+    // Smart Card IdP (X509)
+    piv?: Piv;
+    // Bootstrapping from a recovery token
+    recoveryToken?: string;
+  }
+
+  // Auth params
+  interface AuthParams extends OktaAuthOptions {
+    display?: Display;
+    responseType?: ResponseType | Array<ResponseType>;
+    responseMode?: ResponseMode;
+    scopes?: Array<Scope>;
+    state?: string;
+    nonce?: string;
+    authScheme?: string;
+  }
+  type Display =
+    'popup' |
+    'page';
+  type ResponseMode =
+    'okta_post_message' |
+    'fragment' |
+    'query' |
+    'form_post';
+  type ResponseType =
```

Comment:
Would be easier to make these an enum type?

Reply:
I think for user it would be easier to migrate JS code to TS if using string literals that are documented in readme.
Otherwise with enums user needs to write something like this:
```js
  display: OktaSIW.ResponseMode.OktaPostMessage
```
instead of 
```js
  display: 'okta_post_message'
```
For code editors like VSCode string literals also have autocompletion.
Seems like enums have no real benefit over literal, but I can be wrong..